### PR TITLE
Functional programming section

### PR DESCRIPTION
**Pull request by jotas6**
Content for the section about functional programming of the workshop

## Status
- [x] Edit README file
- [x] Code examples
- [x]  Instructor notes
- [x] Reference sheet
- [x] Schedule

### CODE REVIEW

**Comment by vjd** [comment]
@jotas6 and @storopoli there is no point in giving non pharma examples as people who are learning will not relate to this at all. Across all chapters, we should strive to keep the examples relevant. Get motivation from pumas examples please

**Comment by jotas6** [comment]
Thank you for the review. I will work on making more relevant examples

**Comment by jotas6** [comment]
@storopoli I tried changing the old examples for the more relevant ones that we discussed. I didn't change the ones about `filter` and `count` (https://github.com/PumasAI-Labs/Julia-Workshop/blob/juanjose/functional-programming/04-functional_programming/03-filter.jl) since they already had some pharma context but I could make changes there as well.

File: 04-functional_programming/01-apply.jl [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Functional programming is about creating your programs by applying and composing functions
+
+"""
+A function that calculates the terminal slope using the last and 
+the third to last observations
+"""
+function terminal_slope(observations, time=[0, 1, 2, 4, 8, 12, 24]) # We are setting default time values
```
Comment (storopoli) [inline 04-functional_programming/01-apply.jl:7]:
```suggestion
function terminal_slope(observations; time=[0, 1, 2, 4, 8, 12, 24]) # We are setting default time values
```

**Review by storopoli — APPROVED** [review]
Approving with just a small fix.
Overall I think now that the examples are much more pharmacometrics related.
Thanks!

**Pull request merged**